prusa-slicer: 2.5.2 -> 2.6.0

Moredread opened the pull request:
###### Description of changes

Version bump, also updated wxWidgets to the latest vendored version in the prusa-slicer repo.

A lot of the work was done in https://github.com/NixOS/nixpkgs/issues/222937 by @dmayle. Thanks!

cc @thorstenweber83 

Prebuild binaries are available in the `moredread` cachix cache.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

mweinelt commented on pkgs/applications/misc/prusa-slicer/default.nix:
```diff
@@ -26,28 +29,50 @@
       fetchSubmodules = true;
     };
   });
+  nanosvg-fltk = stdenv.mkDerivation {
+    pname = "nanosvg-fltk";
+    version = "unstable-2022-12-22";
+
+    src = fetchFromGitHub {
+      owner = "fltk";
+      repo = "nanosvg";
+      rev = "abcd277ea45e9098bed752cf9c6875b533c0892f";
+      sha256 = "sha256-WNdAYu66ggpSYJ8Kt57yEA4mSTv+Rvzj9Rm1q765HpY=";
+    };
+
+    nativeBuildInputs = [
+      cmake
+    ];
+  };
+  openvdb_tbb_2021_8 = openvdb.overrideAttrs (old: rec {
+    buildInputs = [ openexr boost tbb_2021_8 jemalloc c-blosc ilmbase ];
+  });
```
I get inlining the `wxGTK-prusa` package, but if you need to package a library, why not in `pkgs/development/libraries`? This also lacks `meta`.

Moredread replied:
I've pulled it out into its own derivation. Is there a nicer way to specify vendored/patched packages?

mweinelt commented on pkgs/applications/misc/prusa-slicer/default.nix:
```diff
@@ -26,28 +29,50 @@
-      rev = "489f6118256853cf5b299d595868641938566cdb";
-      hash = "sha256-xGL5I2+bPjmZGSTYe1L7VAmvLHbwd934o/cxg9baEvQ=";
+      rev = "78aa2dc0ea7ce99dc19adc1140f74c3e2e3f3a26";
+      sha256 = "sha256-rYvmNmvv48JSKVT4ph9AS+JdstnLSRmcpWz1IdgBzQo=";
```
```suggestion
      hash = "sha256-rYvmNmvv48JSKVT4ph9AS+JdstnLSRmcpWz1IdgBzQo=";
```

mweinelt commented on pkgs/applications/misc/prusa-slicer/default.nix:
```diff
@@ -150,7 +165,7 @@ stdenv.mkDerivation rec {
     owner = "prusa3d";
     repo = "PrusaSlicer";
-    sha256 = "sha256-oQRBVAbA2wOYZkQiYIgbd3UcKAkXjnNXo6gB5QbPDAs=";
+    sha256 = "sha256-6AZdwNcgddHePyB0bNS7xGmpz38uzhAwUxgo48OQLuU=";
```
```suggestion
    hash = "sha256-6AZdwNcgddHePyB0bNS7xGmpz38uzhAwUxgo48OQLuU=";
```

mweinelt reviewed (commented):
Can give this a spin later tonight.

Moredread committed:
nanosvg: init at unstable-2022-12-04

Garmelon commented:
When starting `prusa-slicer`, I get a lot of failed assertions along the lines of this one:
![image](https://github.com/NixOS/nixpkgs/assets/11077553/6a3ccc89-7853-406d-b951-060326bf305a)
From what I can tell, these assertions are all GUI related. Not sure how this is related to packaging.

Aside from this, the slicer appears to work (though I've only tested it with a simple model).

Moredread commented:
I get those messages as well (I used an older 'result' path accidentally while testing this PR :( ), it seems there is a PR for this upstream already:

https://github.com/prusa3d/PrusaSlicer/pull/10811

I'll try adding it as a patch for now.

hzeller commented:
LGTM. With the CheckResizer patch, compiled on x86. Successfully used the resulting binary to slice a simple model.

Garmelon commented:
Works well with the patch.

Moredread commented:
I've tested if reenabling the currently disabled tests still increases compiletime/don't work.

With and without those the derivation takes about 15 minutes on my PC, so it looks like we can reenable them.

I'll push the commit later, please wait before merging.

Moredread commented:
I've  disabled tests in slic3r_jobs_tests.cpp as they seem to hang, not always though...

Moredread commented:
> I've disabled tests in slic3r_jobs_tests.cpp as they seem to hang, not always though...

I've reproduced this in Ubuntu Distrobox environment, when running with 'ctest -j', i.e. it looks like a race condition under load. I'll report it upstream asap.

K900 commented:
Are we good to merge here?

SuperSandro2000 commented on pkgs/applications/misc/prusa-slicer/default.nix:
```diff
@@ -21,33 +24,52 @@
+    version = "3.2.0";
+    configureFlags = old.configureFlags ++ [ "--disable-glcanvasegl" ];
+    preConfigure = old.preConfigure + ''
+    sed -ie 's@^\t\(monodll_msw_secretstore\|monolib_msw_secretstore\|basedll_msw_secretstore\|coredll_dark_mode\|monodll_dark_mode\|monolib_dark_mode\|baselib_msw_secretstore\|corelib_dark_mode\)\.o$@\t\1.o \\@' Makefile.in
```
This line looks really unmaintainable. We should move this into a patch

Moredread replied:
I've pulled the patch out.

evils commented on pkgs/development/libraries/nanosvg/default.nix:
```diff
@@ -0,0 +1,23 @@
+  nativeBuildInputs = [ cmake ];
+
+  meta = with lib; {
+    description = "Simple stupid SVG parser ";
```
trailing space here, apparently because their github description has that xD

Moredread replied:
Thanks, corrected it :)

Moredread committed:
prusa-slicer: 2.5.2 -> 2.6.0

Reenabled voronoi tests, as the previous issue of long compile times
seem to have been resolved.

Disabled slic3r_jobs_tests as they fail sometimes.

Co-authored-by: Douglas Mayle <douglas@mayle.org>

Moredread commented:
> Are we good to merge here?

I think after a rebuild with the latest changes we can merge. They shouldn't have changed anything functional.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1009
